client/src/views/Login.tsx

```diff
@@ -40,9 +40,9 @@ export const Login = () => {
         try {
             const token = await AuthService.login({ email, password })
             setAuthStates?.(token)
-
-            navigate(location.state.from || '/')
-        } catch (e: unknown) {
+            navigate(location.state?.from ?? '/')
```

Comment:
Need to make sure state is valid if we're starting from login page

Reply:
This should be ok, only when we have state.from this would be applicable, otherwise redirect to index page.